### PR TITLE
Make nginx listen on IPv6.

### PR DESCRIPTION
**Pull request by ArbitraryCritter**
This makes the nginx proxy listen on IPv6.

It is also possible to make nginx listen with a single listener directive, but I chose this approach because that's exactly how it was done in Loki: https://github.com/grafana/loki/blob/main/production/helm/loki/templates/_helpers.tpl (Search for listen).

I don't believe this warrants a changelog entry, as it only fixes one aspect of IPv6 compatibility.

#### What this PR does
This makes the nginx proxy listen on IPv6 in addition to IPv4.

#### Which issue(s) this PR fixes or relates to

Relates to #3743

#### Checklist

- [X] Tests updated
- [X] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by 56quarters** [comment]
Looks like you'll have to run `make build-helm-tests` and commit the results. I'll let someone more familiar with Helm review but this LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 * [ENHANCEMENT] Query-scheduler: add `cortex_query_scheduler_enqueue_duration_seconds` metric that records the time taken to enqueue or reject a query request. #5879
 * [ENHANCEMENT] Query-frontend: add `cortex_query_frontend_enqueue_duration_seconds` metric that records the time taken to enqueue or reject a query request when not using the query-scheduler. #5879
 * [ENHANCEMENT] Expose `/sync/mutex/wait/total:seconds` Go runtime metric as `go_sync_mutex_wait_total_seconds_total` from all components. #5879
+* [ENHANCEMENT] Nginx: listen on IPv6 addresses. #5948
```
Comment (krajorama) [inline CHANGELOG.md:10]: Sorry, should go into the helm chart changelog under operations/helm/charts/mimir-distributed

Comment (krajorama) [reply CHANGELOG.md:10]: We have some documentation on this as well: https://github.com/grafana/mimir/blob/main/docs/internal/contributing/contributing-to-helm-chart.md  , hope it's useful

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 2]
```diff
@@ -34,6 +34,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 * [ENHANCEMENT] Update the `rollout-operator` subchart to `0.8.0`. #5718
 * [ENHANCEMENT] Make store_gateway podManagementPolicy configurable. #5757
 * [BUGFIX] Fix `global.podLabels` causing invalid indentation. #5625
+* [ENHANCEMENT] Nginx: listen on IPv6 addresses. #5948
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:37]:
```suggestion
* [ENHANCEMENT] Nginx: listen on IPv6 addresses. #5948
* [BUGFIX] Fix `global.podLabels` causing invalid indentation. #5625
```

**Review by krajorama — APPROVED** [review]
LGTM

**Comment by krajorama** [comment]
Cool , only contributor license agreement is left

**Comment by krajorama** [comment]
> @krajorama - I've already signed it. Just re-signed it in case that helped.
> 
> If you check this thread, the CLA assistant reported it as well.

I went into the console eventually and made CLA recheck the PRs.

**Pull request merged**